fbw-common/src/systems/shared/src/GsxSync.ts

```diff
@@ -0,0 +1,407 @@
+    if (!refuel && this.fuelHose.get() === GsxStates.CONNECTED && this.isDefuel) {
+      this.setDesiredFuel(this.desiredFuel);
+      const refuelRate = this.getRefuelRate();
+      this.setRefuelRate(RefuelRate.INSTANT);
+
+      this.fuelTimer.schedule(() => {
+        this.setRefuelStartedState(true);
+        this.isDefuel = false;
+        this.desiredFuel = 0;
+      }, DELAY_REFUEL_RESTART);
+
+      this.rateTimer.schedule(() => {
+        this.setRefuelRate(refuelRate);
+      }, DELAY_RATE_RESTORE);
+    }
+
+    if (!refuel) {
+      this.isRefuelActive = false;
+    }
+  }
+
+  protected evaluateGsxPowerSource(): void {
+    if (!this.isSyncEnabled(SyncServices.POWER)) {
+      return;
+    }
+
+    const extAvail = this.isExtPowerAvail();
+    const jetwayState = this.stateJetway.get();
+    const gpuState = this.stateGpu.get();
+    const boardState = this.stateBoard.get();
+    const departureState = this.stateDeparture.get();
+    let action = PowerAction.NOOP;
+
+    //external is available (as of internal State) - check if disconnect is needed
+    if (extAvail) {
+      //pushback was requested (when boarding not running)
+      if (
+        boardState !== GsxServiceStates.REQUESTED &&
+        boardState !== GsxServiceStates.ACTIVE &&
+        departureState >= GsxServiceStates.REQUESTED
+      ) {
+        action = PowerAction.DISCONNECT;
+        //pushback is active (in it wasn't catched before)
+      } else if (departureState === GsxServiceStates.ACTIVE) {
+        action = PowerAction.DISCONNECT;
+        //jetway and gpu both not connected
+      } else if (
+        jetwayState >= GsxServiceStates.CALLABLE &&
+        jetwayState <= GsxServiceStates.REQUESTED &&
+        gpuState !== GsxServiceStates.ACTIVE
+      ) {
+        action = PowerAction.DISCONNECT;
+      }
+      //external is not available - check if connect is needed
+    } else {
+      if (
+        //jetway or gpu connected
+        (jetwayState === GsxServiceStates.ACTIVE || gpuState === GsxServiceStates.ACTIVE) &&
+        departureState < GsxServiceStates.REQUESTED
+      ) {
+        action = PowerAction.CONNECT;
+      }
+    }
+
+    if (action == PowerAction.CONNECT) {
+      this.setExtPower(true);
+    } else if (action == PowerAction.DISCONNECT) {
+      this.setExtPower(false);
+    }
+  }
+
+  protected isExtPowerAvail(): boolean {
+    let state = false;
+    for (let index = 1; index <= this.numberOfGPUs; index++) {
+      state ||= this.extPowerAvailStates.get(index).get();
+    }
+    return state;
+  }
+
+  protected abstract setOvhdPushBtn(index: number, state: boolean): void;
+
+  protected setExtPower(connect: boolean): void {
+    for (let index = 1; index <= this.numberOfGPUs; index++) {
+      SimVar.SetSimVarValue(`L:A32NX_EXT_PWR_AVAIL:${index}`, SimVarValueType.Bool, connect);
+      if (!connect) {
+        this.setOvhdPushBtn(index, connect);
+      }
+    }
+  }
+}
+
+export class GsxSyncA320 extends GsxSync {
```

Comment:
We don't want to use Airbus trademarks where we can avoid it.
```suggestion
export class GsxSyncA32NX extends GsxSync {
```